python/cudf/cudf/utils/utils.py

```diff
@@ -394,3 +394,33 @@ def pa_mask_buffer_to_mask(mask_buf, size):
         dbuf.copy_from_host(np.asarray(mask_buf).view("u1"))
         return Buffer(dbuf)
     return Buffer(mask_buf)
+
+
+def pos_from_val(n, start, step, length, side="left"):
```

Comment:
I believe this is doing roughly the same thing as `range(start, length, step).index(n)`, which seems more Pythonic. Can we just use that?

Reply:
Yup, did not know that function exists. Will try using that one.

Reply:
Ah but just realized that we might still need this since it doesn't support floats

Reply:
Although we don't seem to be handling floats here...

Reply:
Yeah currently `pos_from_val` only expect integer input, although I think it also handles floats. But I don't think it's being used anywhere in other codes.

Reply:
@shwina so `range.index` seems more restrictive to this method as it requires the argument must present in the range, while this method here will find the closest position to its left/right.

Reply:
Will update the docstring to make that more obvious.

Reply:
Can you provide an example of what doesn't work using `range.index`?

Reply:
Discussed offline and decided to keep the utility (and rename it to something like `search_range()`).